_rules/SC4-1-1-unique-attrs.md

```diff
@@ -0,0 +1,135 @@
+---
+name: attributes are not duplicated
+rule_type: atomic
+description: |
+  This rule checks that HTML and SVG starting tags do not contain duplicated attributes.
+
+success_criterion:
+- 4.1.1 # Parsing
+
+test_aspects:
+- Source code
+
+authors:
+- Wilco Fiers
+- Emma Pratt Richens
+---
+
+
+## Applicability
+
+The rule applies to any [starting tag](https://www.w3.org/TR/html5/syntax.html#start-tags) in an HTML or SVG file.
+
+**Note:** This rule cannot be tested on the DOM Tree because the browser ignores duplicated attributes.
+
+## Expectation
+
+For each test target, there are no duplicated [attributes](https://www.w3.org/TR/html5/syntax.html#elements-attributes).
+
+## Assumptions
+
+*There are currently no assumptions.*
+
+## Accessibility support
+
+*There are no major accessibility support issues known for this rule.*
+
+## Background
+
+- - [H94: Ensuring that elements do not contain duplicate attributes](https://www.w3.org/TR/WCAG20-TECHS/H94.html)
+
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
```

Comment:
Do you need to have test cases repeated...one for each type of attribute for each element type that supports attributes?  What about WAI-ARIA attributes? Is WAI-ARIA out of scope for now?

Reply:
ARIA is irrelevant for this, since that only starts to apply once the DOM is build. This is about the HTML source, not the DOM. Can you clarify what you mean by "each type of attribute"? I suppose we could add attributes with single, double and without quotes into the mix, but that's the only difference I can think of. Can you elaborate what you'd like to see changed (if anything)?

Reply:
Isn't ARIA relevant if there are duplicate ARIA attributes? Only one of the duplicates ends up in the DOM and A11y Tree

```
<input type='text' aria-label='First Name' aria-label='Last Name' />
```

Reply:
And does this apply to all attributes? Or only those with an accessibility impact?

So this is definitely a problem:

```
<img alt='One' alt='Two' src='one.png'> <!-- only one alt attribute ends up in DOM -->
```

but what about this:

```
<img alt='One' alt='One' src='one.png'> <!-- only one alt='One' attribute ends up in DOM -->
```

or this:

```
<meta name='generator' content='Frontweaver 8.2' content='Hugo 0.7'>
```